pkg/vulnsrc/vulnerability/vulnerability.go

```diff
@@ -179,6 +180,34 @@ func getReferences(details map[types.SourceID]types.VulnerabilityDetail) []strin
 	return refs
 }
 
+func getPublishedDate(details map[types.SourceID]types.VulnerabilityDetail) *time.Time {
+	// Despite AllSourceIDs having an order of precedence, ensure we always use NVD first.
+	for _, source := range append([]types.SourceID{NVD}, AllSourceIDs...) {
```

Comment:
```suggestion
	for _, source := range AllSourceIDs {
```

IMO, this should use whatever precedence is defined by `AllSourceIDs` (which is "NVD first" anyway).